addons/website/static/src/components/views/page_views_mixin.js

```diff
@@ -21,18 +21,42 @@ export const PageControllerMixin = (component) => class extends component {
         this.website = useService('website');
         this.dialog = useService('dialog');
         this.rpc = useService('rpc');
+        this.orm = useService("orm");
 
         this.websiteSelection = odoo.debug ? [{id: 0, name: this.env._t("All Websites")}] : [];
 
         this.state = useState({
             activeWebsite: undefined,
+            firstLoad: true,
         });
 
         onWillStart(async () => {
             await this.website.fetchWebsites();
             this.websiteSelection.push(...this.website.websites);
             this.state.activeWebsite = this.website.currentWebsite || this.website.websites[0];
         });
+
+        useEffect(() => {
+            (async () => {
+                const websiteId = this.state.activeWebsite.id;
+                if (!websiteId) {
+                    this.env.searchModel.reload();
+                } else {
+                    const activeWebsitePages = (await this.orm.searchRead(
+                        'website.page',
+                        [['website_id', '=', websiteId]],
+                        ['key']
+                    )).map(rec => rec.key);
+                    const domain = [
+                        '|', ['website_id', '=', websiteId],
+                             '&', ['website_id', '=', null],
+                                  ['key', 'not in', activeWebsitePages],
+                    ];
+                    this.env.searchModel.reload({domain});
+                }
+                this.env.searchModel.search();
+            })();
+        },() => [this.state.activeWebsite]);
```

Comment:
Why do we need this (query again the DB I mean) if we already have all the records behind the scene available? We can't access those from here?

Reply:
From what I understood (but I might be wrong, the ListController and the like are quite intricate, so maybe I overlooked something), we can access the available records, but for all the components (pager, selection, view) to take only the ones we want into consideration, we need to refresh the domain. The whole logic of the controller depends on the search model.
So I'm doing this in a `useEffect()` to update everytime we filter. And I've tried to do it on page load (to avoid the kind of useless `recordFilter()`), but in `onWillStart()`, `this.model.root` isn't set yet, so it's too early, and in the following step it's too late.

I've looked for a better way to do that, but didn't find any. Maybe the filters are tied to an action that should be triggered to update the domain... But if so, which and where?

Reply:
ping @rdeodoo

Reply:
I tried to see with the frameworkjs how it should have been done (in the first place initially, and now as a fix).
I got some information but no real solution for now, I am waiting for their reply

Reply:
Followup: https://github.com/odoo/odoo/pull/148372#issuecomment-1879019780